### PR TITLE
chore: simplify ca cert handling for app and inso

### PR DESCRIPTION
**Pull request by jackkav**
The first request could be getting blocked by file writes, this process is avoided by using the CAINFO_BLOB feature of node-libcurl
- ~~Moves ca cert loading outside of the request path to main app start and inso send-request.~~
- Inso now uses `os.tmpdir()` rather than stubbing electron to point at os.tmpdir()
- considering making all main/nodejs tmpdir calls use `os.tmpdir()` for consistency. 
- NOTE Oauth2 flow still blocking a clear renderer seperation, so we can't remove the getTempDir() helper just yet as react component code paths are still hit by oauth2 re-request.
- blob support was added in https://github.com/JCMais/node-libcurl/pull/300 looks like it should work :rocket:  and allows us to eliminate two file writes!

Future work
- remove electron shim from inso abstraction.
- encourage oauth 2 re-request loop to respect architectural boundaries.
  - remove conditional window/electron and window/require hacks
 
 Closes INS-1511

### CODE REVIEW

File: packages/insomnia/src/network/network.ts [hunk 1]
```diff
@@ -38,7 +38,6 @@ import { isWorkspace } from '../models/workspace';
 import * as pluginContexts from '../plugins/context/index';
 import * as plugins from '../plugins/index';
 import { getAuthHeader } from './authentication';
-import caCerts from './ca_certs';
```
Comment (DMarby) [inline packages/insomnia/src/network/network.ts:41]: Should be able to remove `packages/insomnia/src/network/ca_certs.ts` from `.gitignore`  as well now that this is gone I think?

Comment (jackkav) [reply packages/insomnia/src/network/network.ts:41]: oh nice catch thanks!

File: packages/insomnia/src/main.development.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as electron from 'electron';
 import contextMenu from 'electron-context-menu';
 import installExtension, { REACT_DEVELOPER_TOOLS, REDUX_DEVTOOLS } from 'electron-devtools-installer';
-import { writeFile } from 'fs';
+import {  writeFile } from 'fs/promises';
```
Comment (johnwchadwick) [inline packages/insomnia/src/main.development.ts]: I hate to have a nit this severely minuscule, but I assume you only wanted one space before `writeFile` and not two.

**Review by johnwchadwick — APPROVED** [review]
LGTM, nice. I honestly forgot that we already got the node-libcurl side of things done :)

File: packages/insomnia/src/main.development.ts (hunk 2 above)
Comment (jackkav) [reply packages/insomnia/src/main.development.ts]: thanks, that was an accident

**Review by DMarby — APPROVED** [review]

File: packages/insomnia/package.json [hunk 3]
```diff
@@ -19,14 +19,13 @@
   "scripts": {
     "bootstrap": "npm run build:sr",
     "prebuild": "npm run clean",
-    "build": "npm run generate:ca-certs && npm run build:sr && npm run build:app",
+    "build": "npm run build:sr && npm run build:app",
     "build:app": "esr --cache ./scripts/build.ts --noErrorTruncation",
     "build:main.min.js": "cross-env NODE_ENV=development esr esbuild.main.ts",
-    "build:sr": "npm run generate:ca-certs && esr esbuild.sr.ts",
+    "build:sr": " esr esbuild.sr.ts",
```
Comment (dimitropoulos) [inline packages/insomnia/package.json:25]: @jackkav sorry I didn't catch this in time, but maybe get rid of the extra space before `esr` in a future PR some time if you remember it.

**Pull request merged**